src/commands/lint/mod.rs

```diff
@@ -0,0 +1,118 @@
+pub use args::*;
+#[cfg(feature = "rustup")]
+use serde::Deserialize;
+
+#[cfg(feature = "rustup")]
+use crate::external_cli::Package;
+
+mod args;
+
+/// Represents the contents of `rust-toolchain.toml`.
+#[cfg(feature = "rustup")]
+#[derive(Deserialize)]
+struct RustToolchain {
+    toolchain: Toolchain,
+}
+
+#[cfg(feature = "rustup")]
+#[derive(Deserialize)]
+struct Toolchain {
+    channel: String,
+}
+
+/// Runs `bevy_lint`, if it is installed, with the given arguments.
+///
+/// Calling `lint(vec!["--workspace"])` is equivalent to calling `bevy_lint --workspace` in the
+/// terminal. This will run [`find_bevy_lint()`] to locate `bevy_lint`.
+pub fn lint(args: &mut LintArgs) -> anyhow::Result<()> {
+    use anyhow::ensure;
+
+    #[cfg(feature = "web")]
+    use crate::commands::lint::args::LintSubcommands;
+    use crate::{bin_target::select_run_binary, config::CliConfig, external_cli::cargo};
+
+    let auto_install = args.auto_install();
```

Comment:
(Nit) It feels a bit weird to be calling this so early in the process, since it's not used until `cmd.ensure_status(auto_install)`. Can this just be inlined into `cmd.ensure_status(args.auto_install())`?